qiita_db/support_files/patches/python_patches/14.py

```diff
@@ -0,0 +1,20 @@
+# Feberuary 7, 2015
+# This patch recreates all the QIIME mapping files to avoid lower-upper case
+# problems. See https://github.com/biocore/qiita/issues/799
+
+from qiita_db.util import get_mountpoint
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+_id, fp_base = get_mountpoint('templates')[0]
+
+for prep_template_id in conn_handler.execute_fetchall(
+        "SELECT prep_template_id FROM qiita.prep_template"):
+    prep_template_id = prep_template_id[0]
+    pt = PrepTemplate(prep_template_id)
+    study_id = pt.study_id
+
+    for _, fpt in pt.get_filepaths():
+        pt.create_qiime_mapping_file(fpt)
```

Comment:
If I understand correctly what we would want to do is to "unlink" (just
remove the reference in the DB) the previous QIIME mapping file and then
add this newly created mapping file. As far as I can tell, there's no
way to unlink a QIIME mapping file, @josenavas, can you confirm this? If
that's the case, then I think there are two options (1) add a
`unlink_qiime_mapping_file` method to the object and use that method as
part of the patch or (2) unlink the files with inline code within the
patch (could even be SQL code).

On (Feb-09-15|21:25), Joshua Shorenstein wrote:
> > +from qiita_db.util import get_mountpoint
> > +from qiita_db.sql_connection import SQLConnectionHandler
> > +from qiita_db.metadata_template import PrepTemplate
> > +
> > +conn_handler = SQLConnectionHandler()
> > +
> > +_id, fp_base = get_mountpoint('templates')[0]
> > +
> > +for prep_template_id in conn_handler.execute_fetchall(
> > +        "SELECT prep_template_id FROM qiita.prep_template"):
> > +    prep_template_id = prep_template_id[0]
> > +    pt = PrepTemplate(prep_template_id)
> > +    study_id = pt.study_id
> > +
> > +    for _, fpt in pt.get_filepaths():
> > +        pt.create_qiime_mapping_file(fpt)
> 
> For safety's sake as we go forward, it's probably safer to get rid of the old versions in the DB but still have some way of caching the old ones. Im not sure how often the older SVN versions are gone back to but a similar setup should still exist in the new system as well.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/829/files#r24389882